### ISSUE TITLE
Use weight multiple times in the loss function binary_cross_entropy

### ISSUE THREAD

**Issue by chensnathan** [open-mmlab/mmdetection#845]
The weight has been already used in [this line](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/losses/cross_entropy_loss.py#L36).
Maybe [this line](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/losses/cross_entropy_loss.py#L39) should be:

```
loss = weight_reduce_loss(loss, reduction=reduction, avg_factor=avg_factor)
```

**Comment by hellock** [comment]
Thanks a lot! This should be a potential bug, although currently it will not affect the performance, since all weights are binary values. It leads to unexpected behavior if weights are values other than 0 or 1.

Would you create a PR to fix it?

**Comment by chensnathan** [comment]
Ok!
